spec/rails_helper.rb

```diff
@@ -592,6 +593,20 @@ class SpecTimeoutError < StandardError
         ActiveRecord::Base.connection.schema_cache.add(table)
       end
 
+      ApplicationController.class_eval do
+        before_action do
+          if BlockRequestsMiddleware.current_example_location
+            store_preloaded(
+              "customAjaxRequestHeaders",
+              {
+                "RSPEC_CURRENT_EXAMPLE_LOCATION" =>
+                  BlockRequestsMiddleware.current_example_location,
+              }.to_json,
+            )
+          end
+        end
+      end
```

Comment:
Didn't really want to pollute the `ApplicationController` with something custom like this so hacking this in 😅